### PR TITLE
Reader: Only use post thumbs that appear to be images

### PR DESCRIPTION
**Pull request by blowery**
There are some problems with how these are synced in Jetpack 4.2, so we end up with some odd values from time to time. Add a sanity check that tries to ensure the post thumb is actually an image before using it. Affects Laughing Squid.

See http://calypso.localhost:3000/read/feeds/12098

Test live: https://calypso.live/?branch=fix/reader/post-thumbs-are-sometimes-bad

### CODE REVIEW

File: client/lib/post-normalizer/utils.js [hunk 1]
```diff
@@ -87,3 +85,18 @@ export function domForHtml( html ) {
 	}
 	return dom;
 }
+
+export function thumbIsLikelyImage( thumb ) {
```
Comment (bluefuton) [inline client/lib/post-normalizer/utils.js]: Can we add a method description here?

File: client/lib/post-normalizer/utils.js [hunk 2]
```diff
@@ -36,22 +35,21 @@ export function maxWidthPhotonishURL( imageURL, width ) {
 		return imageURL;
 	}
 
-	let parsedURL = url.parse( imageURL, true, true ), // true, true means allow protocol-less hosts and parse the querystring
-		isGravatar, sizeParam;
+	const parsedURL = url.parse( imageURL, true, true ); // true, true means allow protocol-less hosts and parse the querystring
```
Comment (bluefuton) [inline client/lib/post-normalizer/utils.js]: Like the param description 👍

**Comment by bluefuton** [comment]
Do you have an example of a post this is affecting on Laughing Squid? I can't see any difference in the first couple of pages on https://wpcalypso.wordpress.com/read/feeds/12098.

**Comment by blowery** [comment]
Huh, it's much less pronounced today than yesterday.

Look in the console and you'll see a bunch of 400 errors for various URLs that are trying to load LS pages via photon. You won't see those on the branch.

**Comment by blowery** [comment]
Ah, I see why. There's a check in reader/stream/post.jsx that avoids the canonical image if it's not equal to the featured_image. Yesterday, the featured image and the thumb.URL were equal, so it would override the embed with the feature. Today, the feature is blank, while the thumb is not, so they're not equal and we use the embed.

**Comment by blowery** [comment]
Hrm, this breaks http://calypso.localhost:3000/read/feeds/23259639 compared to wpcalypso.

**Comment by fraying** [comment]
I did notice a bunch of broken images in the stream for Laughing Squid yesterday.

**Comment by blowery** [comment]
> Hrm, this breaks http://calypso.localhost:3000/read/feeds/23259639 compared to wpcalypso.

fixed with https://github.com/Automattic/wp-calypso/pull/7807/commits/4435452bca777e73d7f875dfb5df2bd0b11ba19a

**Pull request merged**